app/integrations/discord/interactions/resolvers/resolver.rb

```diff
@@ -0,0 +1,51 @@
+module Discord
+  module Interactions
+    module Resolvers
+      class Resolver
+        class << self
+          def create_resolver(request:)
+            Resolvers::Ping.new(request:) if request.ping_type?
+          end
+        end
+
+        PUBLIC_KEY = ENV['PUBLIC_KEY']
+        RESPONSE_STATUS = {
+          OK: 200,
+          UNAUTHORIZED: 401
+        }.freeze
```

Comment:
This is knowledge that comes from HTTP specs, we should not let that kind of things go so deep into our code. Dealing with response statuses is a responsibility of the controller